sap/rfc/user.py

```diff
@@ -410,14 +413,15 @@ def create_user(self, connection, user_builder: UserBuilder) -> BAPIReturn:
         """Creates a new user for the given user data"""
 
         rfc_ret = self._call_bapi_method(connection, 'BAPI_USER_CREATE1', user_builder.build_rfc_params())
-        result = rfc_ret['RETURN']
+        bapi_result = BAPIReturn(rfc_ret['RETURN'])
+        if bapi_result.is_error:
+            return bapi_result
 
-        if not BAPIReturn(result).is_error:
-            rfc_password_change_ret = self._change_user_password(connection, user_builder)
-            if rfc_password_change_ret:
-                result.append(rfc_password_change_ret)
+        rfc_password_change_ret = self._change_user_password(connection, user_builder)
+        if rfc_password_change_ret:
+            rfc_ret['RETURN'].append(rfc_password_change_ret)
```

Comment:
Don't you want to call extend( ) instead of append()?

Reply:
No, I don't :-D
`_change_user_password()` returns always only one message. For adding one element into collection `append()` is better as `extend()`. By using `extend()` I would have to convert `rfc_password_change_ret` to a list.